<!doctype html>
<html lang="en">
<link rel="icon" type="image/png" href="/assets/img/favicon.png" >
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="search-domain" value="https://revbayes.github.io/">
    <link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet">
    <link rel="stylesheet" href="/assets/css/syntax.css">
    <link rel="stylesheet" type="text/css" href="/assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="/assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="/assets/css/main.css" />
    <title>RevBayes: Convergence assessment</title>
  </head>
  <body>
    <div class="container">
      <nav class="navbar navbar-default navbar-fixed-top">
  <div class="container-fluid">
    <div class="navbar-header">
      <a href="/" class="pull-left">
        
        <img class="navbar-logo" src="/assets/img/aquabayes-desaturated.png" alt="RevBayes Home" />
        
      </a>
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar" align="right"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">
        <li><a href="/download">Download</a></li>
        <li><a href="/tutorials/">Tutorials</a></li>
        <li><a href="/documentation/">Documentation</a></li>
        <li><a href="https://revbayes.github.io/revscripter/">RevScripter</a></li>
        <li><a href="/workshops/">Workshops</a></li>
        <li><a href="/jobs/">Jobs</a></li>
        <li><a href="/developer/">Developer</a></li>
      </ul>
      <!-- <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form> -->
    </div>
  </div>
</nav>

      <div class="titlebar">
	<h1 class="maintitle">Convergence assessment</h1>
	<h3 class="subtitle">Phylogenetic convergence assessment using the R package Convenience</h3>
	<h4 class="authors">Luiza Fabreti and Sebastian Höhna</h4>
  <h5>Last modified on July  9, 2020</h5>
</div>


<div class="sidebar no-print">
<blockquote class="overview" id="overview">
  <h2>Overview</h2>
  
  <div class="row">
    <div class="col-md-9">
        <strong>Prerequisites</strong>
        
          <ul id="prerequisites">
          
            <li><a href="/tutorials/intro/">Getting Started with RevBayes and Rev Language Syntax</a></li>
          
            <li><a href="/tutorials/mcmc/">Introduction to Markov chain Monte Carlo (MCMC) Sampling</a></li>
          
            <li><a href="/tutorials/ctmc/">Nucleotide substitution models</a></li>
          
            <li><a href="/tutorials/fbd/">Combined-Evidence Analysis and the Fossilized Birth-Death Process for Stratigraphic Range Data</a></li>
          
          </ul>
        
    </div>
  </div>
  
</blockquote>





<blockquote class="tutorial_files" id="tutorial_files">
    <h2>Data files and scripts</h2>
    
        
        <strong>Data Files</strong>
        <ul id="data_files">
        
        
        
          <li><a href="/tutorials/convergence/data/bears_cytb_GTR_run_1.log">bears_cytb_GTR_run_1.log</a></li>
        
          <li><a href="/tutorials/convergence/data/bears_cytb_GTR_run_1.trees">bears_cytb_GTR_run_1.trees</a></li>
        
          <li><a href="/tutorials/convergence/data/bears_cytb_GTR_run_2.log">bears_cytb_GTR_run_2.log</a></li>
        
          <li><a href="/tutorials/convergence/data/bears_cytb_GTR_run_2.trees">bears_cytb_GTR_run_2.trees</a></li>
        
        </ul>
    
</blockquote>


</div>
<h2 class="section" id="overview">Overview</h2>
<hr class="section" />

<p>This tutorial covers convergence assessment of a Bayesian phylogenetic analysis using the R package Convenience.</p>

<p>Convergence of an MCMC analysis is crucial to assure that the chain has sampled from the stationary distribution and that we have sufficiently many samples to approximate the posterior distribution. That is, the MCMC has explored the parameter space long enough to reach the true posterior distribution of the parameters and the values we are sampling belong to that distribution. Theory says that a chain that runs through an infinite time, will reach convergence. For our practical problem, we need to make a decision of when we have sampled enough to take a good estimate of the desired parameters. Here we will show how to test if we have enough samples or not.</p>

<p>An ideal solution would be to analytically calculate the number of steps needed to reach convergence. This, however, has turned out to be unfeasible because the number of samples depends on the specific model, the specific moves applied within the MCMC simulation, and the given dataset.</p>

<p>Since we lack a theoretical convergence assessment, what is broadly done in the MCMC field is analyze the output from the MCMC for lack of convergence.
To do so, we have to keep in mind two aspects of an analysis that has reached convergence: <strong>precision</strong> and <strong>reproducibility</strong>. Precision means that if we run the chain longer, we do not change the estimates (e.g., the posterior mean estimate). While reproducibility means that if we run another independent chain, we get the same estimates.
Precision can be evaluated by checking that we have sufficiently many samples because more samples lead to less variance in the estimates (e.g., the standard error of the mean). Reproducibility, on the other hand, can be evaluated by comparing independent chains run under the same model. Therefore, it’s recommended to run at least two or better four replicates when performing MCMC analyses.</p>

<p>Another best practice is to remove the initial samples from the chain. Those initial iterations are called <em>burn-in</em>. By that we try to get rid of the samples that are not taken from the stationary distribution.</p>

<p>One last concept we need to keep in mind is the <em>Effective Sample Size</em> (ESS), the number of independent samples generated by our MCMC sampler. The ESS takes into account the correlation between samples within a chain. Low ESS values represent high autocorrelation in the chain. If the autocorrelation is higher, then the uncertainty in our estimates is also higher.</p>

<h2 class="section" id="criteria-for-convergence">Criteria for Convergence</h2>
<hr class="section" />

<p>Now that we learned about convergence, let’s take a look into the criteria in the Convenience package:</p>

<p>The output of a phylogenetic analysis most commonly consists of two types of parameters:</p>

<ol>
  <li>Continuous parameters: the evolutionary model parameters, the tree length, clock rates, etc.;</li>
  <li>Discrete parameters: the phylogenetic tree.</li>
</ol>

<p>To assess convergence for these parameters, the Convenience package evaluates:</p>

<ol>
  <li>The Effective Sample Size (ESS);</li>
  <li>Comparison between windows of the same run;</li>
  <li>Comparison between different runs.</li>
</ol>

<p>The comparison between windows of the same run works by dividing the full length of the run into 5 windows (subsets) and comparing the third and fifth window. In the following figure we can see a trace plot for the tree length from the example provided in this tutorial. The trace plot shows the sampled values over the iterations of the MCMC. The highlighted areas of the figure show the third and fifth window of the run.</p>

<figure id="windows"><p><img src="figures/windows.png" /></p>
<figcaption>Trace plot of the tree length for our example analysis. The shaded areas show the third and fifth windows of the run.</figcaption>
</figure>

<h3 class="subsection" id="how-do-we-compare-windows-and-runs">How do we compare windows and runs?</h3>
<hr class="subsection" />

<p>For the continuous parameters, the comparison is made with the two-sample Kolmogorov-Smirnov (KS) test, a non-parametric statistical test for equality of probability distributions. Two samples will be equal when the KS value is below a given threshold. The KS value (D) is calculated:</p>

<script type="math/tex; mode=display">{D}_{m,n} = \max_{x} |{F_{1,m}(x) - G_{2,n}(x)}|</script>

<p>F(x) and G(x) are the empirical distribution functions for the first and second sample, respectively.
The two samples will be drawn from different distributions, at level $\alpha$, when:</p>

<script type="math/tex; mode=display">{D}_{m,n} > c(\alpha) \sqrt{\frac{m + n}{m\times n}}</script>

<p>with</p>

<script type="math/tex; mode=display">c(\alpha) = \sqrt{-\ln({\frac{\alpha}{2})\times \frac{1}{2}}}</script>

<p>The phylogenetic tree is evaluated regarding the bipartitions or splits. Therefore, the comparisons are made using the frequency of a given split between intervals of the same run or between different runs.</p>

<figure id="splits.png"><p><img src="figures/splits.png" /></p>
<figcaption>Two example trees with tips A, B, C, D and the splits seen at each tree.</figcaption>
</figure>

<h3 class="subsection" id="thresholds">Thresholds</h3>
<hr class="subsection" />

<p>The current state of convergence assessment in Bayesian phylogenetics relies mainly on visual tools (e.g., Tracer) and ESS thresholds that have no clear theory to support them <a class="citation" href="#Rambaut2018">(Rambaut et al. 2018)</a>. The motivation for the Convenience package is to provide an easy-to-use framework with clear thresholds for each convergence criterion.</p>

<p>We derive a minimum value for the ESS based on a normal distribution and the standard error of the mean (SEM).
How much error in our estimate of the posterior mean should we find acceptable? Clearly, the mean estimate for a distribution with a large variance does not need to be as precise as the mean estimate for a distribution with a small variance. However, relative to the variance/spread of the distribution, what percentage is acceptable? We suggest to use a SEM smaller of 1% of the 95% probability interval of the distribution, which is equivalent to say that the allowed error of the mean is four times the standard deviation of the distribution. (Note that you can derive a different ESS value for any other threshold that you like.) From this SEM, we can derive the ESS with:</p>

<p>\begin{eqnarray}
SEM = \frac{\sigma}{\sqrt{ESS}}<br />
\frac{\sigma}{\sqrt{ESS}} &lt; 1\% \times 4 \times \sigma<br />
ESS &gt; \frac{1}{0.04^2}<br />
ESS &gt; 625
\end{eqnarray}
<script type="math/tex">SEM = \frac{\sigma}{\sqrt{ESS}}</script>
<script type="math/tex">% <![CDATA[
\frac{\sigma}{\sqrt{ESS}} < 1\% \times 4 \times \sigma %]]></script>
<script type="math/tex">ESS > \frac{1}{0.04^2}</script>
<script type="math/tex">ESS > 625</script></p>

<p>An ESS of 625 is therefore the default value for the convenience package.</p>

<p>For the KS test, the threshold is the critical value for $\alpha$ = 0.01.
We are currently still evaluating the choice of $\alpha$ on the power and false-discovery rate to detect failure of convergence.</p>

<h4 class="subsubsection" id="split-frequncies">Split Frequncies</h4>
<hr class="subsubsection" />

<p>To date, the most often test to assess convergence of split frequencies is the <em>average standard deviation of split frequencies</em> (ASDSF) <a class="citation" href="#Nylander2008">(Nylander et al. 2008)</a>. The frequency of each split is computed for two separate MCMC runs and the difference between the two split frequency estimates is used. The ASDSF is problematic for two reasons: (1) for large trees with many splits that have posterior probabilities close to 0.0 or 1.0 will overwhelm the ASDSF and hence even a single split that is present in all samples in run 1 (thus a posterior probability of 1.0) and is never present in any sample in run 2 (thus a posterior probability of 0.0) might not be detected, and (2) the expected difference in split frequency depends on the true split frequency (see Figure <a href="#difference_split_frequencies"></a>).</p>

<figure id="difference_split_frequencies"><p><img src="figures/difference_split_frequencies.png" width="400" /></p>
<figcaption>The expected difference in split frequencies for ESS of 100, 200 and 625. The x-axis is the true value of the split frequency. The y-axis is the expected difference in split frequencies. The effect of increasing the ESS is the decrease of differences in frequency of sampled splits.</figcaption>
</figure>

<p>Instead of the ASDSF we use the ESS of each split.
We transform each split into a chain of absence and presence values; if the split was present in the i-th tree then we score the i-th value of the chain as a 1 and 0 otherwise. This sequence of absence and presence observations (0s and 1s) allows us to apply standard methods to compute ESS values and thus we can use the same ESS threshold of 625 as for our continuous parameters.</p>

<p>#The threshold for the split frequencies is the mean absolute difference between two variables drawn independently from the same binomial distribution.
#
#<script type="math/tex">{E}[\Delta^{sf}_{p}] = \sum\limits_{i=0}^N \left(|\frac{i}{N} - p| \times P_{binom}(i|N,p) \right)</script></p>

<h2 class="section" id="install">Install</h2>
<hr class="section" />

<p>To install Convenience, we need first to install the package devtools.
In R, type the commands:</p>

<blockquote>
  <p><code class="highlighter-rouge">install.packages("devtools")</code> <br />
<code class="highlighter-rouge">library(devtools)</code> <br />
<code class="highlighter-rouge">install_github("lfabreti/convenience")</code> <br />
<code class="highlighter-rouge">library(convenience)</code> <br /></p>
</blockquote>

<h2 class="section" id="example">Example</h2>
<hr class="section" />

<p>First, download the files listed as data files on the top left of this page. Save them in a folder called data.
These files are the output from a phylogenetic analysis performed with a dataset from bears. The nucleotide substitution model was GTR+$\Gamma$+I and the MCMC was set to run 2 independent runs.
The package also works if your analysis has only one run of the MCMC. But the part to compare runs will not be evaluated. Therefore, it is not possible to say that your MCMC result is reproducible. We strongly advise on running more than 1 run.</p>

<p>The function <code class="highlighter-rouge">checkConvergence</code> takes the output from a phylogenetic analysis performed on RevBayes and works through the convergence assessment pipeline.
This function can take either a directory with all the output files from a single analysis or a list of files.</p>

<p>The function has 3 arguments:</p>
<ol>
  <li><code class="highlighter-rouge">path</code>: for when a path to a directory is provided</li>
  <li><code class="highlighter-rouge">list_files</code>: for when a list of files is provided</li>
  <li><code class="highlighter-rouge">control</code>: a function to set the burnin size, the precision of the standard error of the mean and the continuous parameters to exclude from the assessment. Default values are burnin = 10%, precision = 1%, namesToExclude = “br_lens, bl, Iteration, Likelihood, Posterior, Prior”.</li>
</ol>

<p>Let’s run this function with our example output in a directory(this step may take a few minutes):</p>

<blockquote>
  <p><code class="highlighter-rouge">check_bears &lt;- checkConvergence("data/")</code> <br /></p>
</blockquote>

<p>We can also list the names of the files:</p>

<blockquote>
  <p><code class="highlighter-rouge">check_bears &lt;- checkConvergence( list_files = c("bears_cytb_GTR_run_1.log", "bears_cytb_GTR_run_1.trees", "bears_cytb_GTR_run_2.log", "bears_cytb_GTR_run_2.trees") )</code> <br /></p>
</blockquote>

<p>Now, let’s see what is the output:</p>

<figure id="output_example"><p><img src="figures/output_example.png" /></p>
<figcaption>The output stored in check_bears</figcaption>
</figure>

<p>We can see that <code class="highlighter-rouge">check_bears</code> has 4 elements: <code class="highlighter-rouge">message</code>, <code class="highlighter-rouge">converged</code>, <code class="highlighter-rouge">continuous_parameters</code> and <code class="highlighter-rouge">tree_parameters</code>.</p>

<ol>
  <li><code class="highlighter-rouge">message</code>: a string with a message if the analysis has converged or not;</li>
  <li><code class="highlighter-rouge">converged</code>: a boolean that has TRUE if the analysis converged and FALSE if it did not converge;</li>
  <li><code class="highlighter-rouge">continuous_parameters</code>: a list with ESS, KS-scores between windows of the same run and KS-scores between runs;</li>
  <li><code class="highlighter-rouge">tree_parameters</code>: a list with ESS, split frequencies between windows of the same run and split frequencies between runs.</li>
</ol>

<p>In case the analysis has failed to converge, another element will be on the output list: <code class="highlighter-rouge">failed</code> with the parameters that failed the criteria for convergence.</p>

<p>We can check <code class="highlighter-rouge">continuous_parameters</code>and <code class="highlighter-rouge">tree_parameters</code> with the commands:</p>

<blockquote>
  <p><code class="highlighter-rouge">check_bears$continuous_parameters</code> <br />
<code class="highlighter-rouge">check_bears$tree_parameters</code> <br /></p>
</blockquote>

<figure id="list_parameters"><p><img src="figures/list_parameters.png" /></p>
<figcaption>The lists of what is stored in the <code class="highlighter-rouge">check_bears$continuous_parameters</code> and <code class="highlighter-rouge">check_bears$tree_parameters</code></figcaption>
</figure>

<p>Both lists have 3 elements: ESS, comparison of windows and comparison of runs. Let’s see each component:</p>

<ol>
  <li>Continuous parameters
    <blockquote>
      <p><code class="highlighter-rouge">check_bears$continuous_parameters$ess</code> <br />
<code class="highlighter-rouge">check_bears$continuous_parameters$compare_windows</code> <br />
<code class="highlighter-rouge">check_bears$continuous_parameters$compare_runs</code> <br /></p>
    </blockquote>
  </li>
  <li>Tree parameters
    <blockquote>
      <p><code class="highlighter-rouge">check_bears$tree_parameters$ess</code> <br />
<code class="highlighter-rouge">check_bears$tree_parameters$compare_windows</code> <br />
<code class="highlighter-rouge">check_bears$tree_parameters$compare_runs</code> <br /></p>
    </blockquote>
  </li>
</ol>

<p>Each of these elements can be accessed to check all the values calculated in the package.</p>

<p>Now that we learned how to use the package and how to interpret the results, let’s practice with some exercises.</p>

<h3 class="subsection" id="exercise-1">Exercise 1</h3>
<hr class="subsection" />

<p>Check for convergence in the output generated in the <a href="/tutorials/ctmc/">Nucleotide Substitution Models</a> tutorial.</p>

<p>Which analysis have converged?</p>

<h3 class="subsection" id="convergence-failure">Convergence failure</h3>
<hr class="subsection" />

<p>You can see that the output is different when we have a failure in convergence. We have 2 extra elements <code class="highlighter-rouge">failed</code> and <code class="highlighter-rouge">failed_names</code>.
The element <code class="highlighter-rouge">failed</code> has a summary text of what failed in our analysis.
The element <code class="highlighter-rouge">failed_names</code> has the specifics about the parameters that failed, like the name of the parameter or the split and if it failed in checking for the ESS, the comparison between windows or between runs.</p>

<p><a href="#failed"></a> shows the results from the convergence assessment on the analysis from Exercise 1 with the GTR+$\Gamma$+I nucleotide substitution model.</p>

<figure id="failed"><p><img src="figures/failed.png" /></p>
<figcaption>The output of a convergence assessment that has failed. On the left, we see the output with 6 elements. On the right, it’s the summary of the parameters that failed.</figcaption>
</figure>

<h3 class="subsection" id="exercise-2">Exercise 2</h3>
<hr class="subsection" />

<ul>
  <li>Rerun the MCMC with the GTR+$\Gamma$+I model from <a href="/tutorials/ctmc/">Nucleotide Substitution Models</a>, but increase the number of iterations to 50000.</li>
  <li>Check the new results for convergence.</li>
</ul>

<h3 class="subsection" id="what-to-do">What to do</h3>
<hr class="subsection" />

<figure id="what_to_do" class="table"><figcaption class="table">What to do in case of failure of convergence for each criteria</figcaption>

<table>
  <thead>
    <tr>
      <th style="text-align: left"><strong>Criterion</strong></th>
      <th style="text-align: center"><strong>What we can try</strong></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Combination of criteria</td>
      <td style="text-align: center">Get more samples</td>
    </tr>
    <tr>
      <td style="text-align: left">ESS</td>
      <td style="text-align: center">Get more samples</td>
    </tr>
    <tr>
      <td style="text-align: left">Comparison between windows</td>
      <td style="text-align: center">Increase burn-in</td>
    </tr>
    <tr>
      <td style="text-align: left">Comparison between runs</td>
      <td style="text-align: center">Add more weight to the moves</td>
    </tr>
  </tbody>
</table>
</figure>

<ol class="bibliography"><li><span id="Nylander2008">Nylander J.A.A., Wilgenbusch J.C., Warren D.L., Swofford D.L. 2008. AWTY (are we there yet?): a system for graphical exploration of MCMC convergence in Bayesian phylogenetics. Bioinformatics. 24:581.</span>

</li>
<li><span id="Rambaut2018">Rambaut A., Drummond A.J., Xie D., Baele G., Suchard M.A. 2018. Posterior Summarization in Bayesian Phylogenetics Using Tracer 1.7. Systematic Biology. 67:901–904.</span>

<a href="https://doi.org/10.1093/sysbio/syy032">10.1093/sysbio/syy032</a>

</li></ol>

<script type="text/javascript">
var _ol = document.querySelectorAll('ol');
for (var i = 0, elem_ol; elem_ol = _ol[i]; i++) {
	if ( elem_ol.classList == "bibliography" ) {
		var _li = elem_ol.getElementsByTagName("li");
		//for (var j = 0, elem_li; elem_li = _li[j]; j++)
		//{
		//	elem_li.innerHTML = elem_li.innerHTML.replace(/(https?:\/\/)([^\s<]+)/,"<a href=\"$1$2\">$2");
		//}
		if(_li.length > 0)
			elem_ol.outerHTML = "<h2 class='references'>References</h2><hr class='references'>"+elem_ol.outerHTML
	}
}
</script>

      <br>
<footer>
  <div class="container">
  <div class="row">
    <div class="col-sm-12" align="center">
      <a href="https://github.com/revbayes">GitHub</a> | <a href="/license">License</a> | <a href="/citation">Citation</a> | <a href="https://groups.google.com/forum/#!forum/revbayes-users">Users Forum</a>
    </div>
  </div>
  <br>
  </div>
</footer>

    </div>
    <script src="/assets/js/vendor/jquery.min.js"></script>
<script src="/assets/js/vendor/FileSaver.min.js"></script>
<script src="/assets/js/vendor/jszip.min.js"></script>
<script src="/assets/js/vendor/bootstrap.min.js"></script>

<script type="text/javascript">
// Add default language
$(":not(code).highlighter-rouge").each(function() {
  
  if( this.classList == "highlighter-rouge") {
    this.classList = "Rev highlighter-rouge";
  }
  
});
// $("code.highlighter-rouge").each(function() {
//   
//   if( this.classList == "highlighter-rouge") {
//       this.classList = "Rev highlighter-rouge";
//   }
//   
// });
</script>
<script type="text/javascript"
    src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      processEscapes: true
    }
    });
    MathJax.Hub.Queue(function () {
      $(".aside").each(function() {
          $("div .MathJax", this).hide();
      });
    });
</script>
<script src="/assets/js/base.js"></script>

  </body>
</html>
