<!doctype html>
<html lang="en">
<link rel="icon" type="image/png" href="/assets/img/favicon.png" >
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="search-domain" value="https://revbayes.github.io/">
    <link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet">
    <link rel="stylesheet" href="/assets/css/syntax.css">
    <link rel="stylesheet" type="text/css" href="/assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="/assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="/assets/css/main.css" />
    <title>RevBayes: Bayesian phylogenetics and macroevolution in RevBayes</title>
  </head>
  <body>
    <div class="container">
      <nav class="navbar navbar-default navbar-fixed-top">
  <div class="container-fluid">
    <div class="navbar-header">
      <a href="/" class="pull-left">
        
        <img class="navbar-logo" src="/assets/img/aquabayes-desaturated.png" alt="RevBayes Home" />
        
      </a>
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar" align="right"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">
        <li><a href="/download">Download</a></li>
        <li><a href="/tutorials/">Tutorials</a></li>
        <li><a href="/documentation/">Documentation</a></li>
        <li><a href="https://revbayes.github.io/revscripter/">RevScripter</a></li>
        <li><a href="/workshops/">Workshops</a></li>
        <li><a href="/jobs/">Jobs</a></li>
        <li><a href="/developer/">Developer</a></li>
      </ul>
      <!-- <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form> -->
    </div>
  </div>
</nav>

      <h1>Bayesian phylogenetics and macroevolution in RevBayes</h1>

<b>Location:</b> Room D114, Museum of Paleontology, Richard-Wagner Str 10, LMU Munich, Germany <br>

<b>Dates:</b> 9/17, 2019


to 9/17, 2019


<br>

<b>Instructors:</b> <a href="http://hoehna.org/">Sebastian Höhna</a>, <a href="https://ahsiang.github.io">Allison Hsiang</a>, <a href="https://hoehnalab.github.io/people/">Luiza Fabreti</a> <br>



<h2>Description</h2>

<p>This workshop will introduce participants to the theory and practice of phylogenetic inference methods for macroevolution using RevBayes. The one-day course is part of the "German Paleontological Society Meeting 2019".</p><p>The program for the one-day will entail&#58; (1) An introduction to RevBayes, MCMC, and basic tree inference; (2) Estimating divergence times using molecular clocks; (3) Combining data from fossil and ling taxa. A detailed schedule is given below.</p><p>It is recommended that you review the <i>Pre-workshop Materials</i> listed below if you do not already have a strong understanding of likelihood-based phylogenetics, probability, and Bayesian inference. These video lectures are by <a href="https://phylogeny.uconn.edu/">Paul Lewis</a> and provide a clear overview of these topics. Additionally, a list of papers describing RevBayes and graphical models is provided as suggested background reading.</p>


<h2>Venue</h2> 
Room D114, Museum of Paleontology, Richard-Wagner Str 10, LMU Munich, Germany <br>





<h2>Software</h2>
Before the course starts, please install the following software on your own computer:
<div class="row">
    <div class="col-md-9">
        <ul>
          
            <li><a href="https://revbayes.github.io/revbayes-site/software">RevBayes v1.0.11</a></li>
          
            <li><a href="http://tree.bio.ed.ac.uk/software/tracer/">Tracer</a></li>
          
            <li><a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a></li>
          
            <li>A simple text editor, such as Sublime Text, NotePad++, TextWrangler, BBEdit, vim, or emacs</li>
          
        </ul>
    </div>
</div>



<h2>Pre-workshop Materials</h2>
Please review the following materials before the workshop:
<div class="row">
    <div class="col-md-9">
        <ul>
          
            <li>Paul Lewis's Primer on Phylogenetics Part 1 - <a href="https://youtu.be/1r4z0YJq580">Trees & Likelihood</a></li>
          
            <li>Paul Lewis's Primer on Phylogenetics Part 2 - <a href="https://youtu.be/UsLeY0wZr4Y">Substitution Models</a></li>
          
            <li>Paul Lewis's Primer on Phylogenetics Part 3a - <a href="https://youtu.be/4PWlnNsfz90">Bayesian Statistics & MCMC</a></li>
          
            <li>Paul Lewis's Primer on Phylogenetics Part 3b - <a href="https://youtu.be/TLtOS--YwkU">Bayesian Phylogenetics</a></li>
          
            <li><a href="https://revbayes.github.io/tutorials/intro/getting_started.html">Getting started with <b><i>RevBayes</i></b>.</a></li>
          
            <li><a href="https://revbayes.github.io/tutorials/intro/rev.html">An introductory tutorial to <b>RevBayes and the RevLanguage</b>.</a></li>
          
            <li><a href="https://revbayes.github.io/tutorials/intro/graph_models.html">An introductory tutorial to <b>graphical models</b>.</a></li>
          
        </ul>
    </div>
</div>



<h2>Papers</h2>
We recommend some selected papers for you to read prior to the workshop. These are not required and we will explain all the content throughout the workshop. However, some prior background knowledge will help to successfully work through the material.
<div class="row">
    <div class="col-md-9">
        <ul>
          
            <li><a href="https://academic.oup.com/sysbio/article/63/5/753/2847897">Höhna et al. (2014). Probabilistic Graphical Model Representation in Phylogenetics.</a></li>
          
            <li><a href="https://academic.oup.com/sysbio/article/65/4/726/1753608">Höhna et al. (2016). RevBayes&#58; Bayesian Phylogenetic Inference Using Graphical Models and an Interactive Model-Specification Language.</a></li>
          
            <li><a href="https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/cpbi.22">Höhna et al. (2017). Phylogenetic Inference Using RevBayes.</a></li>
          
        </ul>
    </div>
</div>



<h2>Schedule</h2>






<table class="table table-striped" style="width:100%">
<tr>
<td width="15%"><b>Date</b></td>
<td width="15%"><b>Time</b></td>
<td width="25%"><b>Topic</b></td>
<td width="25%"><b>Material</b></td>
<td width="25%"><b>Instructors</b></td>
</tr>

		<tr>
		<td>Tuesday</td>
		<td>09:30-10:30</td>
    <td>Lecture: Hierarchical Bayesian Modeling for Phylogenetic Inference using RevBayes</td>
		<td>
          
        
        
          
            <a href="https://figshare.com/articles/Hierarchical_Modeling_for_Phylogenetic_Inference_using_RevBayes/7886201">Lecture slides</a>
          
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:30-11:00</td>
    <td>Morning Tea</td>
		<td>
          
        
        
          
            &#x2615;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>11:00-12:00</td>
    <td>Lecture: Introduction to Bayesian Inference and MCMC in RevBayes</td>
		<td>
          
        
        
          
            <a href="https://figshare.com/articles/Introduction_to_Bayesian_Inference_MCMC_in_RevBayes/7886447">Lecture slides</a>
          
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>12:00-13:30</td>
    <td>Lunch</td>
		<td>
          
        
        
          
            &#x1F96A;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>13:30-14:00</td>
    <td>Resources for using RevBayes</td>
		<td>
          
        
        
          
            <a href="https://revbayes.github.io">https://revbayes.com</a>
          
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>14:00-15:30</td>
    <td>Lab: Phylogenetic Inference using RevBayes</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:30-16:00</td>
    <td>Afternoon Tea</td>
		<td>
          
        
        
          
            &#x1F375;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>16:00-17:30</td>
    <td>Lab: Phylogenetic Inference using RevBayes, continued</td>
		<td>
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:30-10:30</td>
    <td>Lecture: Bayesian Divergence-Time Estimation</td>
		<td>
          
        
        
          
            <a href="https://figshare.com/articles/Bayesian_Divergence-Time_Estimation_Lecture/6849005">Lecture slides</a>
          
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:30-11:00</td>
    <td>Morning Tea</td>
		<td>
          
        
        
          
            &#x2615;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>11:30-12:30</td>
    <td>Lecture: Bayesian Divergence-Time Estimation, continued</td>
		<td>
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>12:30-14:00</td>
    <td>Lunch</td>
		<td>
          
        
        
          
            &#x1F35C;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>14:00-15:30</td>
    <td>Lab: Estimating Trees and Divergence Times</td>
		<td>
          
            <a href="/tutorials/fbd/fbd_specimen.html">Combined-Evidence Analysis and the Fossilized Birth-Death Process for Analysis of Extant Taxa and Fossil Specimens</a>
          
        
        
          
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:30-16:00</td>
    <td>Afternoon Tea</td>
		<td>
          
        
        
          
            &#x1F375;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>16:00-17:30</td>
    <td>Lab: Estimating Trees and Divergence Times, continued</td>
		<td>
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>09:30-10:30</td>
    <td>Lab: Historical Biogeography</td>
		<td>
          
            <a href="/tutorials/biogeo/biogeo_simple.html">Simple Phylogenetic Analysis of Historical Biogeography</a>
          
        
        
          
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>10:30-11:00</td>
    <td>Morning Tea</td>
		<td>
          
        
        
          
            &#x2615;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>11:00-12:30</td>
    <td>Lab: Modeling Branch-Heterogeneous Processes</td>
		<td>
        
    </td>
		<td>&#x1F4BB;</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>12:30-14:00</td>
    <td>Lunch</td>
		<td>
          
        
        
          
            &#x1F354;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>14:00-15:30</td>
    <td>Presentations by ANU Researchers</td>
		<td>
        
    </td>
		<td>&#x1F1E6;&#x1F1FA;</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>15:30-16:00</td>
    <td>Afternoon Tea</td>
		<td>
          
        
        
          
            &#x1F375;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>16:00-15:30</td>
    <td>Discussion and Open Lab</td>
		<td>
        
    </td>
		<td>&#x1F5E3;</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>08:00-08:45</td>
    <td>Intro to Bayesian inference</td>
		<td>
          
        
        
          
            <a href="botany2018/lectureBAYESIAN.pdf">Bayesian slides</a><br><a href="botany2018/lectureMCMC.pdf">MCMC slides</a>
          
        
    </td>
		<td>Rosana Zenil-Ferguson</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>08:45-09:00</td>
    <td>Break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>09:00-11:00</td>
    <td>Intro to graphical models and RevBayes</td>
		<td>
          
            <a href="/tutorials/intro/graph_models.html">Introduction to Graphical Models</a>
          
        
        
          
        
    </td>
		<td>Will Freyman</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>11:00-11:15</td>
    <td>Break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>11:15-12:00</td>
    <td>Phylogenetic tree inference</td>
		<td>
          
        
        
          
            <a href="botany2018/lectureCTMC.pdf">CTMC slides</a>
          
        
    </td>
		<td>Rosana Zenil-Ferguson</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>12:00-01:00</td>
    <td>Lunch</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>01:00-03:00</td>
    <td>Phylogenetic tree inference in RevBayes</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Will Freyman</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>03:00-03:10</td>
    <td>Break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>03:10-03:40</td>
    <td>Models linking trait evolution to diversification</td>
		<td>
          
        
        
          
            <a href="botany2018/lectureDIVERSIFICATION.pdf">Diversification slides</a>
          
        
    </td>
		<td>Rosana Zenil-Ferguson</td>
		</tr>
	
		<tr>
		<td>Sunday</td>
		<td>03:40-05:00</td>
    <td>State-dependent speciation and extinction models</td>
		<td>
          
            <a href="/tutorials/sse/bisse.html">State-dependent diversification with BiSSE and MuSSE</a>
          
        
        
          
        
    </td>
		<td>Will Freyman</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:00-09:30</td>
    <td>Welcome to the Workshop</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:30-10:15</td>
    <td>RevBayes Introduction</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:30-12:00</td>
    <td>Substitution Models in Phylogenetics</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>13:00-15:00</td>
    <td>Practical on inferring phylogenies (Part I)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:15-17:30</td>
    <td>Practical on inferring phylogenies (Part II)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:00-09:30</td>
    <td>Recap MCMC and Substitution Models</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:30-10:15</td>
    <td>Morphological models</td>
		<td>
        
    </td>
		<td>Joe O'Reilley, Mark Puttick</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:30-12:00</td>
    <td>Divergence time estimation using the fossilized birth-death model</td>
		<td>
        
    </td>
		<td>Walker Pett</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>13:00-15:00</td>
    <td>Practical on divergence time estimation using the fossilized birth-death model (Part I)</td>
		<td>
          
            <a href="/tutorials/fbd/">Combined-Evidence Analysis and the Fossilized Birth-Death Process for Stratigraphic Range Data</a>
          
        
        
          
        
    </td>
		<td>Walker Pett</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:15-17:30</td>
    <td>Practical on divergence time estimation using the fossilized birth-death model (Part II)</td>
		<td>
          
            <a href="/tutorials/fbd/">Combined-Evidence Analysis and the Fossilized Birth-Death Process for Stratigraphic Range Data</a>
          
        
        
          
        
    </td>
		<td>Walker Pett</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>09:00-09:30</td>
    <td>Recap</td>
		<td>
        
    </td>
		<td>Walker Pett</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>09:30-10:15</td>
    <td>Diversification rate estimation (Part I)</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>10:30-12:00</td>
    <td>Diversification rate estimation (Part II)</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>13:00-15:00</td>
    <td>Choice of practical on diversification rate estimation (Part I)</td>
		<td>
          
            <li><a href="/tutorials/divrate/simple.html">Simple Diversification Rate Estimation</a></li>
          
        
        
          
        
    </td>
		<td>Rachel Warnock, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>15:15-17:30</td>
    <td>Choice of practical on diversification rate estimation (Part II)</td>
		<td>
          
            <li><a href="/tutorials/fbd_range/">Macroevolutionary Analysis of Stratigraphic Range Data</a></li>
          
            <li><a href="/tutorials/divrate/ebd.html">Episodic Diversification Rate Estimation</a></li>
          
            <li><a href="/tutorials/divrate/env.html">Environmental-dependent Speciation & Extinction Rates</a></li>
          
            <li><a href="/tutorials/divrate/branch_specific.html">Branch-Specific Diversification Rate Estimation</a></li>
          
            <li><a href="/tutorials/sse/bisse-intro.html">Background on state-dependent diversification rate estimation</a></li>
          
            <li><a href="/tutorials/sse/bisse.html">State-dependent diversification with BiSSE and MuSSE</a></li>
          
        
        
          
        
    </td>
		<td>Rachel Warnock, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>09:00-10:00</td>
    <td>Lecture: An introduction to Bayesian phylogenetics using morphology</td>
		<td>
        
    </td>
		<td>Mark Puttick</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>10:10-11:00</td>
    <td>Lecture: An introduction to graphical models and Bayesian phylogenetics using RevBayes</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>11:10-12:30</td>
    <td>Lab: Phylogenetic inference using discrete morphology using RevBayes</td>
		<td>
          
            <a href="/tutorials/morph/morph_dec2018.html">Discrete morphology - Tree Inference</a>
          
        
        
          
        
    </td>
		<td>Joseph O'Reilly</td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>09:00-09:30</td>
    <td>Welcome to the Workshop</td>
		<td>
        
    </td>
		<td>Luis Palazzesi</td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>09:30-10:15</td>
    <td>RevBayes Introduction</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>10:30-12:00</td>
    <td>Substitution Models in Phylogenetics</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>13:00-15:00</td>
    <td>Practical on inferring phylogenies (Part I)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna, Luis Palazzesi</td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Monday</td>
		<td>15:15-17:30</td>
    <td>Practical on inferring phylogenies (Part II)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna, Luis Palazzesi</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:00-09:30</td>
    <td>Recap MCMC and Substitution Models</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:30-10:15</td>
    <td>Divergence time estimation</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:30-12:00</td>
    <td>Tip-dating and the fossilized-birth-death model</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>13:00-15:00</td>
    <td>Divergence time estimation</td>
		<td>
          
            <a href="/tutorials/clocks/">Relaxed Clocks & Time Trees</a>
          
        
        
          
        
    </td>
		<td>Femke Thon, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:15-17:30</td>
    <td>Practical on tip-dating and the fossilized-birth-death model</td>
		<td>
          
            <a href="/tutorials/fbd/">Combined-Evidence Analysis and the Fossilized Birth-Death Process for Stratigraphic Range Data</a>
          
        
        
          
        
    </td>
		<td>Femke Thon, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:00-09:30</td>
    <td>Recap</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:30-10:15</td>
    <td>Diversification rate estimation (Part I)</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:30-12:00</td>
    <td>Diversification rate estimation (Part II)</td>
		<td>
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>13:00-15:00</td>
    <td>Practical on diversification rate estimation (Part I)</td>
		<td>
          
            <a href="/tutorials/divrate/simple.html">Simple Diversification Rate Estimation</a>
          
        
        
          
        
    </td>
		<td>Luis Palazzesi, Femke Thon, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:15-17:30</td>
    <td>Practical on diversification rate estimation (Part II)</td>
		<td>
          
            <a href="/tutorials/divrate/branch_specific.html">Branch-Specific Diversification Rate Estimation</a>
          
        
        
          
        
    </td>
		<td>Luis Palazzesi, Femke Thon, Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>09:00-10:00</td>
    <td>Lecture: An Introduction to Bayesian Phylogenetics using RevBayes</td>
		<td>
        
    </td>
		<td>Tracy Heath</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>10:00-11:00</td>
    <td>Lab: Basic MCMC</td>
		<td>
          
            <a href="/tutorials/mcmc/archery.html">Introduction to MCMC using RevBayes</a>
          
        
        
          
        
    </td>
		<td>Tracy Heath</td>
		</tr>
	
		<tr>
		<td>Friday</td>
		<td>11:00-12:00</td>
    <td>Lab: Phylogenetic Inference in RevBayes</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Tracy Heath</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:00-09:30</td>
    <td>Welcome to the Workshop</td>
		<td>
        
    </td>
		<td>Thomas Couvreur, Fabien Condamine</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:30-10:15</td>
    <td>RevBayes Introduction</td>
		<td>
          
        
        
          
            <a href="https://www.dropbox.com/s/pucujjqhw4fknh6/1_RevBayes.pdf?dl=0">RevBayes Intro.</a>
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:30-11:00</td>
    <td>Substitution Models in Phylogenetics</td>
		<td>
          
        
        
          
            <a href="https://www.dropbox.com/s/pswpocjkvd1d16d/2_Bayesian_inference.pdf?dl=0">Bayesian Statistics.</a>
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>11:00-12:00</td>
    <td>Substitution Models in Phylogenetics</td>
		<td>
          
        
        
          
            <a href="https://www.dropbox.com/s/bulg9e6b1muejki/3_Molecular_Evolution_Models.pdf?dl=0">Substitution Models in Phylogenetics</a>
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>13:00-15:00</td>
    <td>Practical on inferring phylogenies (Part I)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:15-17:30</td>
    <td>Practical on inferring phylogenies (Part II)</td>
		<td>
          
            <a href="/tutorials/ctmc/">Nucleotide substitution models</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:00-09:30</td>
    <td>Recap MCMC and Substitution Models</td>
		<td>
          
        
        
          
            <a href="https://www.dropbox.com/s/ayiwjuckxe2vb4r/3_Molecular_Evolution_Models_Recap.pdf?dl=0">Substitution Models in Phylogenetics</a>
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>09:30-10:15</td>
    <td>Introduction to divergence time estimation</td>
		<td>
          
            <a href="/tutorials/dating/">Molecular dating</a>
          
        
        
          
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>10:30-12:00</td>
    <td>Node dating</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>13:00-15:00</td>
    <td>The fossilized birth-death process</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Wednesday</td>
		<td>15:15-17:30</td>
    <td>Total-evidence dating</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>09:00-09:30</td>
    <td>Recap</td>
		<td>
        
    </td>
		<td>Rachel Warnock</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>09:30-10:15</td>
    <td>Diversification rate estimation (Part I)</td>
		<td>
          
            <a href="/tutorials/divrate/simple.html">Simple Diversification Rate Estimation</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>10:15-10:30</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>10:30-12:00</td>
    <td>Diversification rate estimation (Part II)</td>
		<td>
          
            <a href="/tutorials/divrate/ebd.html">Episodic Diversification Rate Estimation</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>12:00-13:00</td>
    <td>Lunch break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>13:00-15:00</td>
    <td>Diversification rate estimation (Part III)</td>
		<td>
          
            <a href="/tutorials/divrate/branch_specific.html">Branch-Specific Diversification Rate Estimation</a>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>15:00-15:15</td>
    <td>Coffee break</td>
		<td>
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Thursday</td>
		<td>15:15-17:30</td>
    <td>Diversification rate estimation (Part IV)</td>
		<td>
          
            <li><a href="/tutorials/sse/bisse-intro.html">Background on state-dependent diversification rate estimation</a></li>
          
            <li><a href="/tutorials/sse/bisse.html">State-dependent diversification with BiSSE and MuSSE</a></li>
          
        
        
          
        
    </td>
		<td>Sebastian Höhna</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>09:40-10:30</td>
    <td>Lecture: Introduction to Bayesian Phylogenetics and RevBayes</td>
		<td>
          
        
        
          
            <a href="https://www.dropbox.com/s/vxk31rdpcyeajkj/1_Bayesian_inference.pdf?dl=0">Lecture slides 1</a> and <a href="https://www.dropbox.com/s/p76sdchegs5taxa/2_RevBayes.pdf?dl=0">Lecture slides 2</a>
          
        
    </td>
		<td>&#x1F468;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>10:30-11:00</td>
    <td>Morning Coffee</td>
		<td>
          
        
        
          
            &#x2615;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>11:00-12:20</td>
    <td>Lecture & Practical: Molecular Clocks</td>
		<td>
          
            <li><a href="/tutorials/dating/global.html">Molecular dating</a></li>
          
            <li><a href="/tutorials/dating/relaxed.html">Molecular dating</a></li>
          
        
        
          
            <li><a href="https://www.dropbox.com/s/qbpoafld1gj5vcx/3_Molecular_Evolution.pdf?dl=0">Lecture slides 3</a> and <a href="https://www.dropbox.com/s/4s1rmkk47vqfl09/4_Clocks.pdf?dl=0">Lecture slides 4</a></li>
          
        
    </td>
		<td>&#x1F468;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>12:20-13:20</td>
    <td>Lunch</td>
		<td>
          
        
        
          
            &#x1F96A;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>13:20-14:40</td>
    <td>Divergence Time Estimation (Time Calibration)</td>
		<td>
          
            <li><a href="/tutorials/dating/nodedate.html">Molecular dating</a></li>
          
            <li><a href="/tutorials/dating/fbdr.html">Molecular dating</a></li>
          
        
        
          
            <li><a href="https://www.dropbox.com/s/k5a6wj6lzbvurb8/5_Time_Calibration.pdf?dl=0">Lecture slides 5</a></li>
          
        
    </td>
		<td>&#x1F468;&#x200D;&#x1F3EB;</td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>14:40-15:10</td>
    <td>Afternoon Coffee</td>
		<td>
          
        
        
          
            &#x2615;
          
        
    </td>
		<td></td>
		</tr>
	
		<tr>
		<td>Tuesday</td>
		<td>15:10-16:30</td>
    <td>Total-Evidence Analysis</td>
		<td>
          
            <li><a href="/tutorials/dating/tefbd.html">Molecular dating</a></li>
          
        
        
          
            <li><a href="https://www.dropbox.com/s/abic5stxas1fxex/6_Total_evidence_dating.pdf?dl=0">Lecture slides 6</a></li>
          
        
    </td>
		<td>&#x1F469;&#x1F3FB;&#x200D;&#x1F3EB;, &#x1F468;&#x200D;&#x1F3EB;</td>
		</tr>
	
</table>








      <br>
<footer>
  <div class="container">
  <div class="row">
    <div class="col-sm-12" align="center">
      <a href="https://github.com/revbayes">GitHub</a> | <a href="/license">License</a> | <a href="/citation">Citation</a> | <a href="https://groups.google.com/forum/#!forum/revbayes-users">Users Forum</a>
    </div>
  </div>
  <br>
  </div>
</footer>

    </div>
    <script src="/assets/js/vendor/jquery.min.js"></script>
<script src="/assets/js/vendor/FileSaver.min.js"></script>
<script src="/assets/js/vendor/jszip.min.js"></script>
<script src="/assets/js/vendor/bootstrap.min.js"></script>

<script type="text/javascript">
// Add default language
$(":not(code).highlighter-rouge").each(function() {
  
  if( this.classList == "highlighter-rouge") {
    this.classList = "default highlighter-rouge";
  }
  
});
// $("code.highlighter-rouge").each(function() {
//   
//   if( this.classList == "highlighter-rouge") {
//       this.classList = "default highlighter-rouge";
//   }
//   
// });
</script>
<script type="text/javascript"
    src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      processEscapes: true
    }
    });
    MathJax.Hub.Queue(function () {
      $(".aside").each(function() {
          $("div .MathJax", this).hide();
      });
    });
</script>
<script src="/assets/js/base.js"></script>

  </body>
</html>
